### PR TITLE
Decide how to trigger unattended upgrades for OCP4

### PR DESCRIPTION
**Pull request by simu**
This isn't final, and may get updated after internal discusssions

### CODE REVIEW

File: docs/modules/ROOT/pages/explanations/decisions/maintenance-trigger.adoc [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+== Rationale
+
+Renovate's approach of creating and automerging MRs to update the desired OpenShift 4 version for individual clusters fits well with our current GitOps approach.
+
+By engineering the upgrade triggering mechanism to interact with the cluster's Project Syn config, we can keep the current machinery based on Project Syn to set the cluster's desired version in the `ClusterVersion` resource unchanged.
+Additionally, we already understand how to engineer extensions for Renovate to create MRs for dependencies which aren't supported out of the box.
+
+Finally, we already have experience with scheduling automerging Renovate MRs for a certain time.
+Since we control Renovate default configuration through self-hosting our custom Renovate, we can set a global default trigger time for upgrades, while still allowing custom start times for tenants or clusters through `renovate.json` in individual tenant repos.
```
Comment (simu) [inline docs/modules/ROOT/pages/explanations/decisions/maintenance-trigger.adoc:59]: Not final yet, but personally, I'm leaning towards Renovate at the moment.

**Review by DebakelOrakel — APPROVED** [review]
Looks good to me

**Comment by anothertobi** [comment]
Although I very much like the idea of triggering updates through Renovate, I fear that this increases complexity a lot by relying on a very specific setup (automated catalog compile and push on commit in a tenant repo).

I'd prefer if an active component in the cluster would update the ClusterVersion object. With this approach we would be more flexible should there be any changes in the future to the update system of OpenShift (per cluster auth against Red Hat API or similar). Monitoring the updates of the ClusterVersion object through an in-cluster component would be fairly easy compared to the alternative of implementing monitoring for failed GitLab jobs.

**Comment by DebakelOrakel** [comment]
> Although I very much like the idea of triggering updates through Renovate, I fear that this increases complexity a lot by relying on a very specific setup (automated catalog compile and push on commit in a tenant repo).
> 
> I'd prefer if an active component in the cluster would update the ClusterVersion object. With this approach we would be more flexible should there be any changes in the future to the update system of OpenShift (per cluster auth against Red Hat API or similar). Monitoring the updates of the ClusterVersion object through an in-cluster component would be fairly easy compared to the alternative of implementing monitoring for failed GitLab jobs.

That means `ocp-drain-monitor` is not `ocp-maint-monitor` but `ocp-maint-operator` :D

**Comment by simu** [comment]
> Although I very much like the idea of triggering updates through Renovate, I fear that this increases complexity a lot by relying on a very specific setup (automated catalog compile and push on commit in a tenant repo).
> 

That's a good point, and as you mention below, monitoring this chain of events to alert an engineer if an update wasn't triggered correctly would be quite complex to implement.

> I'd prefer if an active component in the cluster would update the ClusterVersion object. With this approach we would be more flexible should there be any changes in the future to the update system of OpenShift (per cluster auth against Red Hat API or similar). Monitoring the updates of the ClusterVersion object through an in-cluster component would be fairly easy compared to the alternative of implementing monitoring for failed GitLab jobs.

I think having an active component in the cluster would be a good approach for some aspects of triggering updates, apart from the points you mention, it would also allow us to verify the cluster health before triggering the update. If we go that route, we'd probably want to implement our own controller/operator, since I don't think it would be easy to get the changes we need upstreamed into the openshift-managed-upgrade-operator.

**Comment by simu** [comment]
Updated decision page:
* Added new goals (pre-maintenance health check, maintenance trigger monitoring)
* Added new option to implement our own controller
* Removed decision and rationale for now

**Comment by simu** [comment]
For the review: please check if I forgot to include a point we discussed in the "Rationale" section

**Review by anothertobi — APPROVED** [review]

**Comment by anothertobi** [comment]
The good 10s after approving thoughts: You might want to add that a component running in the cluster fits in well with Red Hat's upgrade concept, which is cluster based.

**Comment by simu** [comment]
> The good 10s after approving thoughts: You might want to add that a component running in the cluster fits in well with Red Hat's upgrade concept, which is cluster based.

Added

**Review by DebakelOrakel — APPROVED** [review]

**Review by corvus-ch — APPROVED** [review]

**Pull request merged**